### PR TITLE
fix Issue 21849 - UTF8: -verrors=context doesn't respect multibyte characters

### PR DESCRIPTION
**Pull request by ibuclaw**
`utf_decodeChar` already deals with incrementing the current byte column position by 1, 2 or 4, so `verrorPrint` doesn't have to worry about printing the correct number of leading spaces counting the number of display columns itself.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: test/fail_compilation/fail21849.d [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// https://issues.dlang.org/show_bug.cgi?id=21849
+// REQUIRED_ARGS: -verrors=context -vcolumns
+/* TEST_OUTPUT:
+---
+fail_compilation/fail21849.d(12,33): Error: cannot implicitly convert expression `cast(ushort)65535u` of type `ushort` to `byte`
```
Comment (ibuclaw) [inline test/fail_compilation/fail21849.d:5]: One enhancement request would be to also put in the display column number `(12, 30)`, not the raw byte column here.  But that would mean fixing `Loc.toChars()` to also be aware of multibyte characters.

**Comment by ibuclaw** [comment]
@thewilsonator added more tests, fixed off-by-one error.
